### DISCUSSION TITLE
jupyterlab myst extension: Render Markup from eval directives

### DISCUSSION THREAD

**Discussion by markuswissinger** [executablebooks/meta#1080] [Ideas]
I would like to have a way for the jupyterlab myst extension to render markup from eval directives. This would allow to create more content dynamically.

### Observed behaviour, kind of a minimal example:

![jupyterlab-myst](https://github.com/executablebooks/meta/assets/18382101/043ff8b4-4ae6-42ee-95cf-c223ef228763)

The markup is not interpreted, the \* characters are printed as plain text.

**Comment by agoose77** [comment]
This is possible if you wrap the text with a display object! e.g.
```python
from IPython.display import Markdown

text_with_markup = Markdown("the **bold** face")
```

**Reply by rowanc1** [reply]
Thanks @agoose77, was going to suggest this as well! @agoose77 @markuswissinger I would be curious on your thoughts on a new role (e.g. `` {md}`myst markdown string` ``) that would parse as MyST?

**Reply by markuswissinger** [reply]
Thank you very much :-)

**Comment by markuswissinger** [comment]
![table_row](https://github.com/executablebooks/meta/assets/18382101/86e69ffb-066b-496c-a5eb-57b4718710dc)

Really appreciate your help. I tried the Markdown class some more. The results are somewhat unexpected.

**Reply by agoose77** [reply]
Hmm. Right now, the `{eval}` role effectively renders the content separately from the rest of the document, and injects the result into the DOM. We could perhaps preferentially render `text/markdown` using MyST directly in this context, but I'm not sure that we could make this particular example work even if that were changed, because the role will be parsed separately from the table. 

To make this example work, you should generate the entire table programmatically, and then display it using `Markdown`.

**Reply by rowanc1** [reply]
Does ``| \{eval}`table_row_content` |`` work? This should parse the table markup as is with a role in it.

[sandbox demo](https://mystmd.org/sandbox?tab=demo&myst=fAAgACAAaABlAGEAZABpAG4AZwAgACAAfAAKAHwALQAtAC0ALQAtAC0ALQAtAC0ALQAtAHwACgB8ACAAewBlAHYAYQBsAH0AYAB4AGAAIAB8AA%3D%3D)

**Reply by markuswissinger** [reply]
I'm a little sad that this kind of templating might be impossible. Might have been quite useful...

- Question that pops up in my mind: how does this approach work with exports to e.g. LaTex? I guess that exporting uses the (html?) DOM only in some special cases.

- In the meantime I tried some more table stuff. Using `Markdown` results in different stylings (font sizes) of text in tables. My guess is the space around the bold text in my post above is also a styling problem. Chasing those on a per case basis might be a rabbit hole not worth going down...

![tables](https://github.com/executablebooks/meta/assets/18382101/67d858d2-5681-4e24-a862-9a0525320529)

- All of the above is absolutely ok for the use case of sharing data and knowledge with collaborators. I don't think it is acceptable for publishing.

While I sound like a harsh critic, I really love what this can do. Its just my use case involves publishing ;-)
